### DISCUSSION TITLE
Mask plugin not working via CDN

### DISCUSSION THREAD

**Discussion by arthurshlain** [alpinejs/alpine#3126] [7. Bugs (archived)] [answered]
I'm trying to run mask directive using CDN, but nothing happens.

```html
<!-- Alpine Plugins -->
<script defer src="https://unpkg.com/@alpinejs/mask@3.x.x/dist/cdn.min.js"></script>

<!-- Alpine Core -->
<script defer src="https://unpkg.com/alpinejs@3.x.x/dist/cdn.min.js"></script>

<p>alpine core test:</p>

<div x-data="{ count: 0 }">
  <button x-on:click="count++">Increment</button>
  <span x-text="count"></span>
</div>

<p>alpine mask plugin test:</p>

<input x-mask="99/99/9999" placeholder="MM/DD/YYYY"> <!-- not working -->
```

**Comment by kukac7** [comment]
hi @arthurshlain 👋 

i think, it works perfect.

[demo](https://jsfiddle.net/kukac7/257pdswf/1/)

**Comment by SimoTod** [comment, marked as answer]
x-data is missing

**Reply by kukac7** [reply]
nope

**Reply by SimoTod** [reply]
Sorry it wasn't meant for you. It was about the original question and example

**Reply by kukac7** [reply]
i thought about the original post too. since the `x-data` is there. 🤔

**Reply by SimoTod** [reply]
It isn't, each directive needs to be defined   on a tag having an ancestor with the x-data attribute (or at least the html tag itself must have x-data). It's what is considered an Alpine component. The fact that there are other tags with x-data that are not ancestor is irrelevant.

**Reply by arthurshlain** [reply]
Thanks. But how it's works in @kukac7 demo without x-data ancestor?

**Reply by SimoTod** [reply]
I don't know, I don't use jsfiddle but that is the behaviour on a real page.
